internal/tracker/bundle_info.go

```diff
@@ -71,31 +72,56 @@ func newBundleInfo(ctx context.Context, ref image.ImageReference, tasks commonTa
 	return &info, nil
 }
 
-// updateCommonTasks updates the commonPipelineTasks attributes with the tasks found
-// in the given pipeline.
-func (info *bundleInfo) updateCommonTasks(ctx context.Context, pipelineName string) error {
+// getBundlePipeline gets a pipeline from a bundle by the pipeline name
+func (info *bundleInfo) getBundlePipeline(ctx context.Context, pipelineName string) (error, v1beta1.PipelineObject) {
 	client := NewClient(ctx)
 	bundle := info.ref.String()
 	runtimeObject, err := client.GetTektonObject(ctx, bundle, "pipeline", pipelineName)
 	if err != nil {
-		return err
+		return err, nil
 	}
 	pipelineObject, ok := runtimeObject.(v1beta1.PipelineObject)
 	if !ok {
-		return fmt.Errorf("pipeline resource, %q, cannot be converted to a PipelineObject", pipelineName)
+		return fmt.Errorf("pipeline resource, %q, cannot be converted to a PipelineObject", pipelineName), nil
 	}
 	pipelineObject.SetDefaults(ctx)
 
+	return nil, pipelineObject
+}
+
+// updateRequiredTasks updates the required common tasks and required pipeline tasks
+func (info *bundleInfo) updatePipelineTasks(ctx context.Context, pipelineName string) error {
+	err, pipelineObject := info.getBundlePipeline(ctx, pipelineName)
+	if err != nil {
+		return err
+	}
 	// Filter out unwanted pipelines
 	// TODO: Consider making this filter configurable at some point in the future.
 	if val, ok := pipelineObject.PipelineMetadata().Labels["skip-hacbs-test"]; ok && val == "true" {
 		return nil
 	}
+	pipelineTaskNames := getTaskNames(pipelineObject.PipelineSpec())
 
-	info.commonPipelineTasks.updateTasksIntersection(getTaskNames(pipelineObject.PipelineSpec()))
+	// set the required pipeline tasks if the proper key is returned
+	if taskKey := getRequiredTaskKey(pipelineObject); taskKey != "" {
+		info.pipelineTasks = map[string][]string{taskKey: sets.List(pipelineTaskNames)}
+	}
+
+	info.commonPipelineTasks.updateTasksIntersection(pipelineTaskNames)
 	return nil
 }
 
+// get the key used in "pipeline-required-tasks" from the pipelineObject
+func getRequiredTaskKey(pipelineObject v1beta1.PipelineObject) string {
```

Comment:
Perhaps this should return a `bool` and be named `holdsRequiredTasks`?

Reply:
So just check for the existence of the key? Where does it get the label value then?

Reply:
I could have missed other uses of `getRequiredTaskKey`; I was thinking about the way the function is used on line 106, basically as a conditional for selecting pipeline tasks, so reflecting that in the name/signature would be helpful.